src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -239,6 +237,26 @@ int CodegenLLVMVisitor::get_num_elements(const ast::IndexedName& node) {
     return static_cast<int>(*macro->get_value());
 }
 
+bool CodegenLLVMVisitor::is_kernel_function(const std::string& function_name) {
```

Comment:
Just as a note : do you think we should add some property / meta data for our kernel function to find kernels directly? Or, we can store their names? I am mentioning this because finding kernels by return type and arg type is not that robust. 

If you think that is worth doing in future, just add `\todo` comment.

Reply:
I think so! I think I mentioned that partly when discussing GPU code generation.

> I am mentioning this because finding kernels by return type and arg type is not that robust.

I absolutely agree! I guess a flag `isKernel` can be sufficient - but then we need to have metadata/attributes on the AST or symbol level. I guess storing the names during AST transformations in the helper visitor would be the best option?

I will add a todo for the function as well (saying it's a temporary thing).